translator/src/main/java/com/google/devtools/j2objc/translate/Autoboxer.java

```diff
@@ -243,6 +244,17 @@ public void endVisit(CastExpression node) {
       if (typeUtil.isAssignable(exprType, typeUtil.getJavaNumber().asType())) {
         // Casting a Number object to a primitive, convert to value method.
         unbox(expr, (PrimitiveType) castType);
+      } else if (exprType == typeUtil.boxedClass(typeUtil.getChar()).asType()) {
+        // Unboxing and casting Character, which does not have number value functions
+        unbox(expr);
+        if (castType.getKind() != TypeKind.CHAR) {
+          // If the resulting type is not char - keep the cast, to preserve type information in
+          // case of reboxing
```

Comment:
Sounds good, thanks for the clarification. We're big fans of explicit code like this rather than detail-hiding wrapper methods, so don't worry about the cast.